src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,119 @@
 public class StringManipulation implements StringManipulationInterface {
-
+	String str = null;
     @Override
     public String getString() {
-        return null;
+        return str;
     }
 
     @Override
     public void setString(String string) {
+    	str = string;
     }
 
     @Override
     public int count() {
-        return 0;
+    	//checks for null or 'empty' one char strings, anything more is a bit beyond scope of project
+    	if( str == null || str == " ") {
+    		return 0;
+    	}
+        int n = 1;
+        for(int i = 1; i < str.length()-1; i++) {
+        	//adds a word count at a space, but doesn't count up for multiple spaces in a row, or a space at the end of a string.
+        	if(str.charAt(i) == ' ' && str.charAt(i-1) != ' ') {
+        		n++;
+        	}
+        }
+        return n;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+    	//exception check
+    	String output = "";
+    	if(n <= 0) {
+    		throw new IllegalArgumentException("n is less than or equal to zero");
+    	}
+    	if( n > str.length()) {
+    		throw new IndexOutOfBoundsException("n is greater than string length");
+    	}
+    	//iterates over string in increments of n, replacing those words, using substring function of String to grab the other snippets. 
+    	for(int i = n; i <= str.length(); i+=n) {
+    		output += str.substring(i-n, i-1);
+    		if(maintainSpacing) {
+    			output += " ";
+    		}
+    	}
+    	//Grabs the remaining portion of the string after the last removed character.
+    	int remain = str.length()%n;
+    	if(remain != 0) {
+    		output += str.substring(str.length() - remain);
+    	}
+        return output;
     }
 
     @Override
-    public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+    public String[] getSubStrings(int startWord, int endWord){
```

Comment:
Regarding Exception Handling: This method is doing a great job of validating its input parameters and throwing exceptions accordingly. However, the error messages could be made a little more user-friendly, perhaps by including the invalid value in the message itself for quick reference.

On String Concatenation: In the loop where words are constructed using result[words] += str.charAt(x);, it's important to remember that strings in Java are immutable and that each concatenation results in the creation of a new object. This could potentially be inefficient in terms of memory usage, especially for long strings. Consider using a StringBuilder for constructing the words, which would be more efficient.

On Loop Control Variable 'x': The variable 'x' is being used to track the index in the original string. It would be helpful to provide a more descriptive name for this variable to make the code more readable.

On Variable Naming: The variables startWord, endWord, and words are well named and convey their purpose effectively. However, x and i could be renamed to something more descriptive to further improve readability. For example, x could be renamed to charIndex and i could be renamed to wordIndex. This could help other developers understand the code more quickly.